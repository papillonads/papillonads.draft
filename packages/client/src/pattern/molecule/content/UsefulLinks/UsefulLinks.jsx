import React from 'react'
import cn from 'classnames'
import cs from './UsefulLinks.scss'
import { usefulLinksDefaultProps, usefulLinksPropTypes } from './UsefulLinks.props'

const UsefulLinks = ({ className, data }) => {
  const usefulLinksClassList = cn(className, cs.usefulLinks)

  return <div className={usefulLinksClassList} />
}

UsefulLinks.defaultProps = usefulLinksDefaultProps

UsefulLinks.propTypes = usefulLinksPropTypes

export default UsefulLinks
