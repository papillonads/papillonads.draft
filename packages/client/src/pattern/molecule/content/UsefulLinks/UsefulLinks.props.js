import { string, shape, arrayOf } from 'prop-types'

export const usefulLinksDefaultProps = {
  className: '',
}

export const usefulLinksPropTypes = {
  className: string,
  data: shape({
    buyersLinks: shape({
      heading: shape({
        title: string,
      }),
      links: arrayOf(
        shape({
          title: string,
          description: string,
          icon: string,
        }),
      ),
    }),
    sellersLinks: shape({
      heading: shape({
        title: string,
      }),
      links: arrayOf(
        shape({
          title: string,
          description: string,
          icon: string,
        }),
      ),
    }),
  }).isRequired,
}
