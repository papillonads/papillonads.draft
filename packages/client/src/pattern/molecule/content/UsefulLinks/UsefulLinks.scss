/* stylelint-disable font-family-no-missing-generic-family-keyword */
.title {
  margin-bottom: 10px;
  font-family: BreeSerif;
  font-weight: 300;
  font-size: 18px;
  line-height: 1.2em;
}

.buyersLinks {
  &Link {
    display: flex;

    &Icon {
      width: 36px;
      height: 36px;
      margin-bottom: 20px;
      align-self: center;
    }

    &Text {
      margin-bottom: 16px;
      line-height: 1.2em;
      cursor: pointer;

      &Title {
        color: #116db4;
        font-weight: 500;
        margin-bottom: 5px;
      }

      &Description {
        color: #2d3c4d;
        font-weight: 300;
      }
    }
  }
}

.sellersLinks {
  background-color: transparent;
}
/* stylelint-enable font-family-no-missing-generic-family-keyword */
