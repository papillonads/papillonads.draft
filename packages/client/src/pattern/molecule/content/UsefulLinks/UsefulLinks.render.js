import React from 'react'
import UsefulLinks from './UsefulLinks'
import homePageData from '../../../../../../server/src/data/page/home'

const defaultRender = <UsefulLinks data={homePageData.content.usefulLinks} />

export { defaultRender }
