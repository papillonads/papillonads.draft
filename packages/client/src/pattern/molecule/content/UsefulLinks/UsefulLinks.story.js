import { storiesOf } from '@storybook/react'
import { defaultRender } from './UsefulLinks.render'
import { getInfo } from '../../../../../../../.storybook/utility'

const defaultRenderInfo = getInfo('default')

storiesOf('Patttern/Molecule/Content/UsefulLinks', module).add(defaultRenderInfo.text, () => defaultRender, defaultRenderInfo.parameters)
