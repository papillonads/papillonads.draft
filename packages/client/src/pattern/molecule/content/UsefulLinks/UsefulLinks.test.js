import { defaultRender } from './UsefulLinks.render'

describe('<UsefulLinks />', () => {
  describe('Snaphot', () => {
    test('must match defaultRender', () => {
      expect(global.renderToJSON(defaultRender)).toMatchSnapshot()
    })
  })
})
