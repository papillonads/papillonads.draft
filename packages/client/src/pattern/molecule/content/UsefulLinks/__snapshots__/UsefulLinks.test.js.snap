// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UsefulLinks /> Snaphot must match defaultRender 1`] = `
<div
  className="usefulLinks"
/>
`;
